### ISSUE TITLE
[Core] Implement Rust Unit Tests for Math Logic

### ISSUE THREAD

**Issue by debug-soham** [etsi-ai/etna#27]
### Description
The core mathematical engine in `etna_core` currently lacks source-level unit tests. Since backpropagation and weight updates are implemented manually, we need to ensure these calculations are correct to prevent silent training failures.

### Target Files
- `etna_core/src/layers.rs`
- `etna_core/src/optimizer.rs`
- `etna_core/src/loss_function.rs`

### Tasks
- [ ] Add `#[cfg(test)]` modules to the bottom of target files.
- [ ] **Linear Layer Test**: Verify that a 2x2 Identity weight matrix with zero bias returns the input exactly.
- [ ] **ReLU Test**: Ensure negative values are zeroed and positive values remain unchanged.
- [ ] **Backprop Test**: Manually calculate the gradient for a single 1x1 Linear layer and assert that `Linear::backward` produces the same result.
- [ ] **Optimizer Test**: Assert that `SGD` updates a weight of `1.0` to `0.99` given a gradient of `0.1` and learning rate of `0.1`.

### Acceptance Criteria
- `cargo test` passes for all modules.
- Code coverage for mathematical functions reaches >90%.

**Comment by magic-peach** [comment]
@debug-soham 
I’d like to take this issue and add source-level unit tests to validate the core mathematical operations in etna_core.

My approach would be:

**Test Setup**

Add `#[cfg(test)] mod tests` blocks at the bottom of:

```
- layers.rs
- optimizer.rs
- loss_function.rs
```

- **Use deterministic, minimal inputs to keep tests simple and reproducible.**

**Linear `Layer` Forward Test**

- Initialize a `Linear laye`r with a 2×2 identity weight matrix and zero bias.

- Pass a known input vector and assert that the output exactly matches the input.

**ReLU Activation Test**

- Provide a fixed input containing both negative and positive values.
- Assert that negative values are zeroed and positive values remain unchanged.

**Backpropagation Test**

- Use a minimal 1×1 `Linear` layer with known weight and bias.
- Manually compute expected gradients for a single forward/backward pass.
- Assert that Linear::backward produces gradients matching the hand-calculated values.

**Optimizer (SGD) Test**

- Initialize a parameter with value `1.0.`
- Apply a gradient of `0.1 `with a learning rate of `0.1.`
- Assert that the updated weight equals `0.99.`

**Validation**

- Run cargo test to ensure all tests pass.
- Verify that mathematical functions achieve `>90%` coverage as required.

I’ll keep each test focused, minimal, and well-commented to make the math explicit and prevent silent training regressions.
